### PR TITLE
Crawl reports listening ports of inbound peers (RIPD-866)

### PR DESCRIPTION
**Pull request by vinniefalco**
This improves the information returned by /crawl

### CODE REVIEW

**Comment by mtrippled** [comment]
:+1:

File: src/ripple/overlay/impl/OverlayImpl.cpp [hunk 1]
```diff
@@ -595,12 +595,21 @@ OverlayImpl::crawl()
             pv[jss::public_key] = beast::base64_encode(
                 sp->getNodePublic().getNodePublic().data(),
                     sp->getNodePublic().getNodePublic().size());
+            pv[jss::type] = sp->slot()->inbound() ?
```
Comment (rec) [inline src/ripple/overlay/impl/OverlayImpl.cpp]: Can `sp->slot()` be null?

File: src/ripple/peerfinder/impl/SlotImp.h [hunk 2]
```diff
@@ -146,6 +160,9 @@ class SlotImp : public Slot
     beast::IP::Endpoint m_remote_endpoint;
     boost::optional <beast::IP::Endpoint> m_local_endpoint;
     boost::optional <RipplePublicKey> m_public_key;
+    // ports are uint16_t values, but we use uint32_t so we can
```
Comment (rec) [inline src/ripple/peerfinder/impl/SlotImp.h]: Consider

    static std::int32_t const noPort = -1;
    std::atomic <std::int32_t> m_listening_port = noPort;

and then earlier:

    if (value == noPort)

**Comment by nbougalis** [comment]
Left a couple of comments. It's ok as is, but I'd prefer to get those addressed.

File: src/ripple/peerfinder/impl/SlotImp.h (hunk 2 above)
Comment (vinniefalco) [inline src/ripple/peerfinder/impl/SlotImp.h]: That's a good idea

File: src/ripple/overlay/impl/OverlayImpl.cpp (hunk 1 above)
Comment (vinniefalco) [inline src/ripple/overlay/impl/OverlayImpl.cpp]: Nope, its an invariant that all fully constructed PeerImp objects have a Slot.

**Comment by nbougalis** [comment]
:+1: from me.

**Comment by nbougalis** [comment]
Looks reasonable.

**Comment by seelabs** [comment]
:+1: for me. Nice changes @vinniefalco

**Pull request closed without merging**